src/main/java/veneto/parser/Parser.java

```diff
@@ -0,0 +1,95 @@
+package veneto.parser;
+
+import veneto.command.*;
+import veneto.exceptions.VenetoException;
+import veneto.task.Deadline;
+import veneto.task.Event;
+import veneto.task.Task;
+import veneto.task.ToDo;
+
+public class Parser {
+    /* Fields */
+    public static final int FIND_COMMAND_INDEX = 5;
+    public static final int TODO_DESCRIPTION_INDEX = 5;
+    public static final int DEADLINE_DESCRIPTION_INDEX = 9;
+    public static final int DEADLINE_START_INDEX = 3;
+    public static final int EVENT_DESCRIPTION_INDEX = 5;
+    public static final int EVENT_START_INDEX = 5;
+    public static final int EVENT_END_INDEX = 3;
+
+    /* Methods */
+    /**
+     * translate user put to commands that need to operate
+     * @param text the input from user
+     * @return the Command that the user wants to operate
+     */
+    public static Command translateCommand(String text) {
+        String[] texts = text.split(" ");
+        String command = texts[0].toLowerCase();
+        switch (command) {
+            case "bye":
+                return new ExitCommand();
+            case "list":
+                return new ListCommand();
+            case "mark":
+                return new MarkCommand(Integer.parseInt(texts[1]), 1);
+            case "unmark":
+                return new MarkCommand(Integer.parseInt(texts[1]), 0);
+            case "todo":
+                return prepareAdd(text, 1);
+            case "deadline":
+                return prepareAdd(text, 2);
+            case "event":
+                return prepareAdd(text, 3);
+            case "delete":
+                return new DeleteCommand(Integer.parseInt(texts[1]));
+            case "find":
+                return prepareFind(text);
+            default:
+                throw new VenetoException("Invalid Command");
+        }
+    }
+
+    /**
+     * generate FindCommand according to the user input
+     * @param text the details about the task
+     * @return the command that the user want to operate
+     */
+    private static Command prepareFind(String text) {
+        String keyword = text.substring(FIND_COMMAND_INDEX);
+        return new FindCommand(keyword);
+    }
+
+    /**
+     * generate AddCommand according to the user input
+     * @param text the details about the task
+     * @param funcId indicates the type of task
+     * @return the command that the user want to operate
+     */
+    private static Command prepareAdd(String text, int funcId) {
+        String[] texts = text.split("/");
+        for (int i = 0; i < texts.length; i++) {
+            texts[i] = texts[i].trim();
+        }
+        Task newTask = null;
+        String description;
+        switch (funcId) {
+            case 1:
```

Comment:
Hi perhaps instead of calling it case 1,2,3 you could name the cases such that it is easier for others to understand which cases this are?